tests/framework/e2e/cluster.go

```diff
@@ -491,3 +493,25 @@ func (epc *EtcdProcessCluster) WithStopSignal(sig os.Signal) (ret os.Signal) {
 	}
 	return ret
 }
+func (epc *EtcdProcessCluster) Leader(ctx context.Context) (EtcdProcess, error) {
```

Comment:
nit: empty line to separate from import. 

What warries me is that this method is relatively 'heavy' (3 connections made)
and it's easy to write very slow test that recaluculates 'leader' before each iteration. 

Mitigations: 
  1. call it 'FindLeaderHeavy' 
  2. cache the leader as part of EtcdProcessCluster and have 2 methods:  `Leader()` that `RefreshLeader()`. 
      `Leader` would only call `RefreshLeader` when there is no populated cache...

Unit test owner would need to call `RefreshLeader` in all the cases when leader is expected to change.

Reply:
I think `heavy` here is very relative. Creating 3 connections could be indeed considered heavy, however it is still lighter then all other methods that create whole process.

As this function is just needed for this test (we are looking for log generated by leader), I have removed it from Cluster.